platform/darwin/asset_root.mm

```diff
@@ -2,13 +2,22 @@
 
 #include <mbgl/platform/platform.hpp>
 
+@interface MGLApplicationRootBundleCanary : NSObject
+@end
+
+@implementation MGLApplicationRootBundleCanary
+@end
+
 namespace mbgl {
 namespace platform {
 
 // Returns the path to the root folder of the application.
 const std::string &assetRoot() {
     static const std::string root = []() -> std::string {
-        NSString *path = [[[NSBundle mainBundle] resourceURL] path];
+        NSString *path = [[NSBundle bundleForClass:[MGLApplicationRootBundleCanary class]] pathForResource:@"MapboxGL" ofType:@"bundle"];
+        if (!path) {
+            path = [[[NSBundle mainBundle] resourceURL] path];
+        }
```

Comment:
We did used to have something like this before, a fallback for our own test app. We should probably do something really drastic here to warn the user, since frequently we will see support around people not having included the bundle properly. But we need to strike a balance between this and not hampering our test app, either. 

Basically, "when present" should be 100% of the time when used as a third-party library, and we should scream about it a little more when this is not the case.